swagger_server/controllers_local/collectioninstrument.py

```diff
@@ -0,0 +1,300 @@
+##############################################################################
+#                                                                            #
+#   Collection Instruments Upload                                            #
+#   Date:    11 May 2017                                                     #
+#   Author:  Gareth Bult                                                     #
+#   License: MIT                                                             #
+#   Copyright (c) 2017 Crown Copyright (Office for National Statistics)      #
+#                                                                            #
+##############################################################################
+from ..configuration import ons_env
+from ..models_local.instrument import InstrumentModel
+from ..models_local.exercise import ExerciseModel
+from ..models_local.business import BusinessModel
+from ..models_local.survey import SurveyModel
+from ..models_local.classification import ClassificationModel, classifications
+from traceback import print_exc
+from sys import stdout
+from json import loads
+from uuid import UUID
+
+DEFAULT_SURVEY = "3decb89c-c5f5-41b8-9e74-5033395d247e"
+
+
+def protect(uuid=True):
+    """
+    This decorator is designed to add a protective layer around endpoint logic, specifically
+    it checks for 'id_example' and allows it for integration tests to succeed. It also ensures that
+    passed parameters are strings, maybe not needed but prevents something poking a dict or similar
+    through by mistake. It then converts the first parameter to a UUID type if the 'uuid' flag is set.
+    
+    :param uuid: Whether to convert first parameter to UUID() 
+    :return: status code, return message
+    """
+    def protect_wrapped(func):
+        def func_wrapped(*args, **kwargs):
+            try:
+                my_id = args[1]
+                if my_id == 'id_example':
```

Comment:
The code would (could) never do this as id's are validated, if someone was hacking an endpoint, it would give them a 200 instead of 204 .. I'm not too worried about a hacker being given the wrong status code .. indeed it might frustrate him for a while .. :)